app/src/main/java/com/mathewsachin/fategrandautomata/util/ImageLoader.kt

```diff
@@ -57,14 +60,29 @@ class ImageLoader @Inject constructor(
 
         if (!regionCachedPatterns.containsKey(path)) {
             val pattern =
-                createPattern("$currentGameServer/${path}")
+                loadPatternWithFallback(path)
 
             regionCachedPatterns[path] = pattern
         }
 
         return regionCachedPatterns[path]!!
     }
 
+    /**
+     * When image is not available for the current server, use the image from NA server.
+     */
+    private fun loadPatternWithFallback(path: String): IPattern {
+        if (currentGameServer != GameServerEnum.En) {
+            return try {
+                createPattern(currentGameServer, path)
+            } catch (e: FileNotFoundException) {
```

Comment:
I don't really like the catch because it's bad for the performance, here's a blog article about C#: https://www.c-sharpcorner.com/blogs/impact-of-trycatch-on-performance

The alternative involves going through all assets in the gameserver folder:
```
if (context.assets.list("$currentGameServer").contains("$path")) {
  //load from specific gameserver folder
}
else {
  //load from NA
}
```

Reply:
I don't think it really matters, since we do it only once per asset and then cache it.
But, yeah, I agree with the suggestion, but you'll have to open a separate PR. I already merged this 😅.